internal/markdown/markdown.go

```diff
@@ -102,24 +102,44 @@ func FromMessages(client *slackclient.SlackClient, history *slackclient.HistoryR
 		return msgTimes[messages[i].Ts].Before(msgTimes[messages[j].Ts])
 	})
 
-	for _, message := range messages {
-		var username string
-		var err error
-
-		if message.User != "" {
-			username, err = client.UsernameForID(message.User)
-		} else if message.BotID != "" {
-			username = fmt.Sprintf("bot %s", message.BotID)
-		} else {
-			username = "ghost"
-		}
+	lastSpeakerID := ""
+
+	for i, message := range messages {
+		username, err := client.UsernameForMessage(message)
 		if err != nil {
 			return "", err
 		}
 
-		fmt.Fprintf(b, "> **%s** at %s\n>\n",
-			username,
-			msgTimes[message.Ts].Format("2006-01-02 15:04"))
+		speakerID := message.User
```

Comment:
What do you think about just using `username` here? Are you concerned about re-use of usernames? If it does happen I am not sure that it'll help users because the only difference in display will be that we won't include an extra speaker header.

Reply:
I figure the ID is unique and we don't need anything human-friendly in this value, just an identifier. Is there a reason to use the username over the ID?

Reply:
It's not a big deal, happy to leave it as you prefer. I suggested it because I thought we could get rid of the `speakerID` variable, the assignment of which duplicates some of the logic from the `usernameForMessage`.

Reply:
Hmm, that's a good point, I'll try it!